### PR TITLE
Create RepositoryBranchesClient

### PR DESCRIPTION
**Pull request by ryangribble**
In preparation for #1407 this PR creates a new "Branches" client as a child of "Repositories", inline with the GitHub API layout.  Existing branch related methods are moved under this client, with the old methods marked as `[Obsolete]`

`client.Repository.GetBranch()` => `client.Repository.Branch.Get()`
`client.Repository.GetAllBranches()` => `client.Repository.Branch.GetAll()`
`client.Repository.GetBranch()` => `client.Repository.Branch.Edit()`

Once this is merged, I will work on the new branch protection related methods for #1407

### CODE REVIEW

**Comment by ryangribble** [comment]
Looks like all the tests passing, just needs some  👀 from someone before merging

**Comment by shiftkey** [comment]
@ryangribble added to my list

**Comment by BigAlInTheHouse** [comment]
Get in contact with me when you're ready thks

File: Octokit/Clients/RepositoriesClient.cs [hunk 1]
```diff
@@ -541,69 +552,65 @@ public Task<IReadOnlyList<Repository>> GetAllForOrg(string organization, ApiOpti
...
-        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
-        /// <returns>All <see cref="T:Octokit.Branch"/>es of the repository</returns>
+        [Obsolete("Please use RepositoriesClient.Branch.GetAll() instead.  This method will be removed in a future version")]
         public Task<IReadOnlyList<Branch>> GetAllBranches(string owner, string name)
         {
             Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
             Ensure.ArgumentNotNullOrEmptyString(name, "name");
-
-            return GetAllBranches(owner, name, ApiOptions.None);
+
```
Comment (alfhenrik) [inline Octokit/Clients/RepositoriesClient.cs]: Unnecessary whitespace 😀

**Comment by alfhenrik** [comment]
Other than some superfluous whitespace, this looks good to me 👍

**Comment by ryangribble** [comment]
whitespace fixed, thanks 😀

**Comment by alfhenrik** [comment]
Running through the integration tests for this locally right now, all 💚  so far 😁
